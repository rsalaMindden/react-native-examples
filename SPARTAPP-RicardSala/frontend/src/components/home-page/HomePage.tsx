import React, {useEffect, useState} from 'react';
import {
  Image,
  SafeAreaView,
  TextInput,
  Text,
  StyleSheet,
  ImageBackground,
  Pressable,
  View,
  TouchableOpacity,
} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import {loadTrainings} from '../../redux/actions/trainingCreators';

import globalStyles from '../../theme/globalTheme';
import buttonStyles from '../../theme/buttonsTheme';


const Home = ({navigation}: any) => {
  const dispatch = useDispatch();
  const trainings = useSelector((store: any) => store.trainings);
  const [search, setSearch] = useState('');
  useEffect(() => {
    dispatch(loadTrainings());
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  
  const trainingFiltereds = trainings.filter(
    (element: any) => element.name === search,
    
  );

  return (
    <SafeAreaView style={[globalStyles.backgound, globalStyles.aliginItems]}>
      <Image
        style={globalStyles.logo__Standar}
        source={require('../../images/spartappYELLOW.png')}
      />

      <TextInput
        style={[styles.search__box, globalStyles.text__gray]}
        placeholder="BUSCAR"
        autoCapitalize="characters"
        placeholderTextColor="#FFD523"
        textAlign='center'
        onChangeText={(text: any) => setSearch(text)}
      />

      <Image
        style={styles.search__icon}
        source={require('../../images/magnifyingglass.png')}
      />
      {search.length < 1 ? (
        <View>
          <ImageBackground
            source={require('../../images/trainingImg.png')}
            resizeMode="cover"
            style={styles.trainingTypes__image}>
            <Pressable onPress={() => navigation.push('popUpTraining')}>
              <Text
                style={[
                  globalStyles.text__yellow,
                  globalStyles.login__text,
                  styles.trainingTypes__text,
                ]}>
                TIPOS DE ENTRENAMIENTO
              </Text>
            </Pressable>
          </ImageBackground>

          <ImageBackground
            source={require('../../images/cronoImg.png')}
            resizeMode="cover"
            style={styles.trainingTypes__image}>
            <Pressable onPress={() => navigation.push('popUpCrono')}>
              <Text
                style={[
                  globalStyles.text__yellow,
                  globalStyles.login__text,
                  styles.trainingTypes__text,
                ]}>
                OBJETIVO POR TIEMPO
              </Text>
            </Pressable>
          </ImageBackground>

          <ImageBackground
            source={require('../../images/equipmentImg.png')}
            resizeMode="cover"
            style={styles.trainingTypes__image}>
            <Pressable onPress={() => navigation.push('popUpEquipment')}>
              <Text
                style={[
                  globalStyles.text__yellow,
                  globalStyles.login__text,
                  styles.trainingTypes__text,
                ]}>
                EQIPAMIENTO
              </Text>
            </Pressable>
          </ImageBackground>
        </View>
      ):<View>
        {trainingFiltereds.map((train:any)=>(
          <View key={train._id}  >
   <TouchableOpacity
    style={[buttonStyles.filter__box ,globalStyles.aliginItems]}
    onPress={() => navigation.push('Training')}
    >
   <Text style={[globalStyles.text__yellow,globalStyles.login__text]}  >
   {train.name}
   </Text>
   </TouchableOpacity>
   
 </View>
        ))}
      </View>
      }
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  search__box: {
    fontWeight: '600',
    fontSize: 24,
    alignContent: 'center',
    backgroundColor: '#2C2E43',
    marginTop: 35,
    width: 206,
    height: 41,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 7,
    },
    shadowOpacity: 0.43,
    shadowRadius: 9.51,

    elevation: 15,
    borderRadius: 15,
  },
  result__box: {
    justifyContent:'center',
    fontWeight: '600',
    fontSize: 24,
    alignContent: 'center',
    backgroundColor: '#2C2E43',
    marginTop: 35,
    width: 300,
    height: 41,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 7,
    },
    shadowOpacity: 0.43,
    shadowRadius: 9.51,

    elevation: 15,
    borderRadius: 15,
  },

  trainingTypes__text: {
    marginTop: 25,
    alignSelf: 'center',
  },
  trainingTypes__image: {
    width: 349,
    height: 155,
    marginBottom: 15,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 7,
    },
    shadowOpacity: 0.43,
    shadowRadius: 9.51,

    elevation: 15,
  },
  search__icon: {
    top: -32,
    right: 75,
  },
});

export default Home;
