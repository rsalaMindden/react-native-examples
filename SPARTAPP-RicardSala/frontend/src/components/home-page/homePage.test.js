import React from "react";
import HomePage from './HomePage'
import {render,fireEvent}from '../../utils/tetsUtils/test.utils'

const navigation={
    pop:jest.fn(),
    push:jest.fn()
}
jest.mock('../../redux/actions/actionCreators',()=>({
    registerUser: jest.fn()
  }));



describe('Given a HomePage component',()=>{
    describe('When it is render',()=>{
        let screen
        beforeEach(()=>{
            screen=render(<HomePage navigation={navigation}/>)
        })
        test('Then should match the snapshot',()=>{
            expect(screen).toMatchSnapshot();
          })

          describe('And popUpEquipment button is clicked', () => {
            test('Then navigation.push should have been called with handleClick', () => {
              const BUTTON = screen.getByTestId('popUpEquipment');
              fireEvent.press(BUTTON);
              expect(navigation.push).toHaveBeenCalled();
            });
          });
          describe('And popUpCrono button is clicked', () => {
            test('Then navigation.push should have been called with handleClick', () => {
              const BUTTON = screen.getByTestId('popUpCrono');
              fireEvent.press(BUTTON);
              expect(navigation.push).toHaveBeenCalled();
            });
          });
          describe('And popUpTraining button is clicked', () => {
            test('Then navigation.push should have been called with handleClick', () => {
              const BUTTON = screen.getByTestId('popUpTraining');
              fireEvent.press(BUTTON);
              expect(navigation.push).toHaveBeenCalled();
            });
          });

        })
      })
  