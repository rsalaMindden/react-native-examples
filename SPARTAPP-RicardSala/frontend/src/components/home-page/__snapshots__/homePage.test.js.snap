// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a HomePage component When it is render Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#595260",
        "flex": 1,
      },
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/images/spartappYELLOW.png",
      }
    }
    style={
      Object {
        "height": 31,
        "marginTop": 54,
        "width": 146,
      }
    }
  />
  <TextInput
    autoCapitalize="characters"
    onChangeText={[Function]}
    placeholder="BUSCAR"
    placeholderTextColor="#FFD523"
    style={
      Array [
        Object {
          "alignContent": "center",
          "backgroundColor": "#2C2E43",
          "borderRadius": 15,
          "elevation": 15,
          "fontSize": 20,
          "fontWeight": "600",
          "height": 41,
          "marginTop": 35,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 7,
            "width": 0,
          },
          "shadowOpacity": 0.43,
          "shadowRadius": 9.51,
          "width": 206,
        },
        Object {
          "color": "#B2B1B9",
        },
      ]
    }
    textAlign="center"
  />
  <Image
    source={
      Object {
        "testUri": "../../../src/images/magnifyingglass.png",
      }
    }
    style={
      Object {
        "right": 75,
        "top": -32,
      }
    }
  />
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "elevation": 15,
            "height": 155,
            "marginBottom": 35,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 7,
              "width": 0,
            },
            "shadowOpacity": 0.43,
            "shadowRadius": 9.51,
            "width": 310,
          }
        }
      >
        <Image
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../src/images/trainingImg.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 155,
                "width": 310,
              },
              Object {
                "borderRadius": 15,
              },
            ]
          }
          testID="popUpTraining"
        />
        <Text
          style={
            Array [
              Object {
                "color": "#FFD523",
              },
              Object {
                "fontSize": 20,
                "fontWeight": "600",
                "justifyContent": "center",
                "marginLeft": 10,
                "marginRight": 10,
              },
              Object {
                "alignSelf": "center",
                "marginTop": 25,
              },
            ]
          }
        >
          TIPOS DE ENTRENAMIENTO
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "elevation": 15,
            "height": 155,
            "marginBottom": 35,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 7,
              "width": 0,
            },
            "shadowOpacity": 0.43,
            "shadowRadius": 9.51,
            "width": 310,
          }
        }
      >
        <Image
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../src/images/cronoImg.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 155,
                "width": 310,
              },
              Object {
                "borderRadius": 15,
              },
            ]
          }
          testID="popUpCrono"
        />
        <Text
          style={
            Array [
              Object {
                "color": "#FFD523",
              },
              Object {
                "fontSize": 20,
                "fontWeight": "600",
                "justifyContent": "center",
                "marginLeft": 10,
                "marginRight": 10,
              },
              Object {
                "alignSelf": "center",
                "marginTop": 25,
              },
            ]
          }
        >
          OBJETIVO POR TIEMPO
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "elevation": 15,
            "height": 155,
            "marginBottom": 35,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 7,
              "width": 0,
            },
            "shadowOpacity": 0.43,
            "shadowRadius": 9.51,
            "width": 310,
          }
        }
      >
        <Image
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../src/images/equipmentImg.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 155,
                "width": 310,
              },
              Object {
                "borderRadius": 15,
              },
            ]
          }
          testID="popUpEquipment"
        />
        <Text
          style={
            Array [
              Object {
                "color": "#FFD523",
              },
              Object {
                "fontSize": 20,
                "fontWeight": "600",
                "justifyContent": "center",
                "marginLeft": 10,
                "marginRight": 10,
              },
              Object {
                "alignSelf": "center",
                "marginTop": 25,
              },
            ]
          }
        >
          EQIPAMIENTO
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
